src/nyc_trees/apps/core/mail/backends/boto.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+import boto
+import json
+import logging
+import traceback
+
+from django.conf import settings
+from django.core.mail.backends.base import BaseEmailBackend
+
+from django_statsd.clients import statsd
+
+logger = logging.getLogger(__name__)
+
+
+class EmailBackend(BaseEmailBackend):
+
+    def __init__(self, fail_silently=False, aws_access_key_id=None,
+                 aws_secret_access_key=None, aws_region=None):
+        super(EmailBackend, self).__init__(fail_silently=fail_silently)
+        self.aws_access_key_id =\
+            aws_access_key_id or settings.EMAIL_AWS_ACCESS_KEY_ID
+        self.aws_secret_access_key =\
+            aws_secret_access_key or settings.EMAIL_AWS_SECRET_ACCESS_KEY
+        self.aws_region =\
+            aws_region or settings.EMAIL_AWS_REGION
+
+    def open(self):
+        try:
+            if not self.connection:
+                self.connection = boto.ses.connect_to_region(
+                    self.aws_region,
+                    aws_access_key_id=self.aws_access_key_id,
```

Comment:
If the node this code runs on has IAM role support, the connection can just be:

```python
self.connection = boto.ses.connect_to_region(self.aws_region)
```

Boto will automatically hit the metadata API behind the scenes and pull in the `aws_access_key_id` and `aws_secret_access_key`: https://github.com/boto/boto/blob/bbbf9d248856db997080f51805575f0634da4090/boto/provider.py#L392-L402

We can add support for providing the credentials explicitly, but the `connect_to_region` would need to branch two ways:

- If credentials are provided, use them
- If no credentials are provided, don't supply the arguments and let Boto try to get them from the metadata